modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -164,6 +166,13 @@ private List<String> findRoutingPaths(String indexName, Settings allSettings, Li
             for (var fieldMapper : mapperService.documentMapper().mappers().fieldMappers()) {
                 extractPath(routingPaths, fieldMapper);
             }
+            for (var objectMapper : mapperService.documentMapper().mappers().objectMappers().values()) {
+                if (objectMapper instanceof PassThroughObjectMapper passThroughObjectMapper) {
+                    if (passThroughObjectMapper.containsDimensions()) {
+                        routingPaths.add(passThroughObjectMapper.fullPath() + ".*");
```

Comment:
Would this include non keyword fields as well? I think it does, but only keyword fields would dynamically be mapped as dimension (in `DynamicFieldsBuilder`).

Reply:
That's correct, I think the contract for the "passthrough" object is that it creates alias at the root level. Dimensions are optional, and we'll likely have options for metrics soon.

Reply:
Updated the javadoc to clarify that eligible non-keyword fields are still marked as dimensions.